high_card.rb

```diff
@@ -1,5 +1,52 @@
-# Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+require "pry"
 
-players = []
+def build_deck
+  ranks = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A" ]
+  suits = [ "hearts", "spades", "clubs", "diamonds" ]
+  deck = []
+  ranks.each_with_index do |rank, i|
+    suits.each do |suit|
+      deck << [rank, suit, i]
+    end
+  end
+  return deck.shuffle
+end
+
+def player_names
+  players = []
+  while true
+    puts "Enter player name or enter 'play' to continue"
+    player = gets.chomp
+      if player == "play"
+        break
+      end
+    players << player
+  end
+  return players
+end
+
+def play
+  deck = build_deck
+  players = player_names
+  hands = []
+  players.each do |player|
+    hand = {
+      player: player, card: deck.pop
+    }
+    hands << hand
+  end
+  return hands
```

Comment:
I like how you are being explicit here with your returns, just know that the last line of each method is implicitly returned in Ruby